### ISSUE TITLE
Bug in hue/saturation and color_temperature code

### ISSUE THREAD

**Issue by avadiax** [mihai-dinculescu/tapo#225]
```rust
/// Sets the *hue* and *saturation*. ['ColorLightSetDeviceInfoParams::send'] must be called at the end to apply the changes.
/// The device will also be turned *on*, unless ['ColorLightSetDeviceInfoParams::off'] is called.
///
/// # Arguments
///
/// * 'hue' - between 1 and 360
/// * 'saturation' - between 1 and 100
pub fn hue_saturation(mut self, hue: u16, saturation: u8) -> Self {
    self.hue = Some(hue);
    self.saturation = Some(saturation);
    self.color_temperature = Some(0);

    self
}

/// Sets the *color temperature*. ['ColorLightSetDeviceInfoParams::send'] must be called at the end to apply the changes.
/// The device will also be turned *on*, unless ['ColorLightSetDeviceInfoParams::off'] is called.
///
/// # Arguments
///
/// * 'color_temperature' - between 2500 and 6500
pub fn color_temperature(mut self, value: u16) -> Self {
    self.hue = Some(0);
    self.saturation = Some(100);
    self.color_temperature = Some(value);

    self
}
```     

Referring to the above code snippet in 'tapo/src/requests/set_device_info/color_light.rs' between line 65 to 92.

The code is wrong because hue must be 1 to 360, but in color_temperature and certain colors in COLOR_MAP, it is hard_coded to 0. Also, there are bulb color settings where the hue, saturation and color_temperature can co-exist such as color temperature is "3300", hue is "10", saturation is "19". Using the command:

> await device.set().brightness(80).hue_saturation(10, 19).color_temperature(3300).send(device)

... will always trigger a Validation error.

>    await device.set_hue_saturation(hue, saturation)
>Exception: Validation { field: "hue", message: "must be between 1 and 360" }

Please confirm if this is the case. Correct fix is that in 'fn hue_saturation', the line 'self.color_temperature' should be removed. In 'fn color_temperature', the lines 'self.hue' and 'self.saturation' should be removed.

**Comment by mihai-dinculescu** [comment]
You're correct, `hue` should accept a value of `0`. I've fixed this in the referenced commit.

However, I do not think the API should allow setting both `color_temperature` and the `hue` and `saturation` pair. Whenever `color_temperature` is set, `hue` and `saturation` seem to have no effect.

Getting the device info confirms this. `hue` and `saturation` will be returned as `None` when set together with `color_temperature`.

**Comment by avadiax** [comment]
Thanks for the confirmation.

With regards to setting the hue/saturation and color_temperature concurrently, I noticed one very interesting thing.

If you refer to screenshot, you see that in the Tapo app, I have configured the following color temp.

![CT3390](https://github.com/mihai-dinculescu/tapo/assets/22733368/a89153fd-9380-43ad-9b1d-ad6d8e7861d3)

When I use device_info to extract the information, this is what I get.

```
Brightness: 90
Color Temperature: 3390
Hue: 45
Saturation: 26
Device On: True
``` 
It is because it returns hue, saturation, and color_temperature values that I thought they can all be set without any issues.

As I dug deeper, I realised that all colors in the "Color" tab will always return a color_temperature of 0, but in the "White Light" tab, depending on which color_temperature chosen, the hue/saturation will return with different values.

The reason I wanted to capture these values and set them is because I am using the bulb as a notification when a state occurs, and the bulb will show a color for a period of time, and *restore* back with the original color setting.

```python
    device = await client.l530(ip_address)

    device_info = await device.get_device_info()
    brightness = device_info.to_dict()['brightness']
    color_temp = device_info.to_dict()['color_temp']
    hue = device_info.to_dict()['hue']
    saturation = device_info.to_dict()['saturation']
    device_on = device_info.to_dict()['device_on']

    await device.set().brightness(80).color(Color.ForestGreen).send(device)
    await asyncio.sleep(2)
    await device.set().brightness(brightness).hue_saturation(hue, saturation).color_temperature(color_temp).send(device)
``` 

If the API forbade or ignore hue/saturation when color_temperature is used, then I will need to rewrite the code to ignore hue/saturation when color_temperature is bigger than 0.

Thanks for looking into this, and look forward to the release of the new version.
